pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -28,34 +28,41 @@ import (
 	v1 "k8s.io/api/core/v1"
 	v1beta1ext "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
 // Plugin is a plugin driver that dispatches containers to each node,
 // expecting each pod to report to the master.
 type Plugin struct {
-	Definition      plugin.Definition
-	DfnTemplateData *plugin.DefinitionTemplateData
-	cleanedUp       bool
+	Definition plugin.Definition
+	SessionID  string
+	Namespace  string
+	cleanedUp  bool
 }
 
 // Ensure DaemonSetPlugin implements plugin.Interface
 var _ plugin.Interface = &Plugin{}
 
+type templateData struct {
+	PluginName        string
+	ResultType        string
+	SessionID         string
+	Namespace         string
+	ProducerContainer string
+	MasterAddress     string
+}
+
 // NewPlugin creates a new DaemonSet plugin from the given Plugin Definition
 // and sonobuoy master address
-func NewPlugin(namespace string, dfn plugin.Definition, cfg *plugin.WorkerConfig) *Plugin {
+func NewPlugin(dfn plugin.Definition, namespace string) *Plugin {
```

Comment:
I wonder if we should continue to allow for namespace overrides.  Does that even make sense?  
We originally (knob'd everything) but I'm now re-evaluating the usefulness.

Reply:
I think it makes sense to parameterise these even if we don't expose them to end users. That way we can have a single source of truth for the namespace

Reply:
I believe namespaces is how heptiocloud allows users to run heptio-e2e and e2es simultaneously.

We probably want to revisit this, but that is how it was set up.